File: frontend/src/objects/player/shop/shop-item.ts
import { z } from "zod";

export const ShopItemSchema = z.object({
  id: z.string().min(1),
  name: z.string().min(1),
  description: z.string().min(1),
  price: z.number().int().min(1),
  currency: z.enum(["coins", "gems"]),
  catalogIndex: z.number().int().min(0),
  active: z.boolean(),
  sortOrder: z.number().int(),
  createdAt: z.string().min(1),
  updatedAt: z.string().min(1),
});

export type ShopItem = z.infer<typeof ShopItemSchema>;

File: backend/microservice/src/player/objects/shop/ShopItem.scala
package microservice.player.objects.shop

import io.circe.generic.semiauto._
import io.circe.{Decoder, Encoder}

/** 商店商品领域对象（玩家浏览与管理员维护共用）。 */
final case class ShopItem(
  id: String,
  name: String,
  description: String,
  price: Int,
  currency: String,
  catalogIndex: Int,
  active: Boolean,
  sortOrder: Int,
  createdAt: String,
  updatedAt: String
)

/** ShopItem 伴生对象：Circe JSON 编解码，供商店 API 响应。 */
private[player] object ShopItem {
  implicit val encoder: Encoder[ShopItem] = deriveEncoder
  implicit val decoder: Decoder[ShopItem] = deriveDecoder
}
